radis/tools/slit.py

```diff
@@ -583,16 +635,50 @@ def convolve_with_slit(
     # --------------
 
     # Assert slit function is thin enough
-    try:
-        assert abs(w[-1] - w[0]) > abs(w_slit[-1] - w_slit[0])
-    except AssertionError:
-        raise AssertionError(
-            "Slit function is broader ({0:.1f}nm) than spectrum".format(
-                abs(w_slit[-1] - w_slit[0])
-            )
-            + " ({0:.1f}nm). No valid range.".format(abs(w[-1] - w[0]))
+
+    def lin_interp(x, y, i, half):
+        return x[i] + (x[i+1] - x[i]) * ((half - y[i]) / (y[i+1] - y[i]))
+
+    def half_max_x(x, y):
+        half = max(y)/2.0
+        signs = np.sign(np.add(y, -half))
+        zero_crossings = (signs[0:-2] != signs[1:-1])
+        zero_crossings_i = np.where(zero_crossings)[0]
+        return [lin_interp(x, y, zero_crossings_i[0], half),
+                lin_interp(x, y, zero_crossings_i[1], half)]
+    def fwhm(x,y):
+        hmx = half_max_x(x,y)
+        return hmx[1] - hmx[0]
+
+    w_range = w[-1] - w[0]
+    w_slit_range = w_slit[-1] - w_slit[0]
+    slit_FWHM = fwhm(w_slit, I_slit)
```

Comment:
RADIS already has a [get_FWHM()](https://github.com/radis/radis/blob/develop/radis/tools/slit.py#L682) function, we could use it directly if it also works as you expect (less code = easier to maintain): 

```suggestion
    w_range = abs(w[-1] - w[0])
    w_slit_range = abs(w_slit[-1] - w_slit[0])
    slit_FWHM = get_FWHM(w_slit, I_slit)
```

(note : you can commit the suggestion directly from the GitHub interface)

Reply:
So I replied in the wrong thread apparently.  :(

Reply:
See 
b9abfe1

Solves the problem